datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -92,24 +109,115 @@ func (f *ChainDataFetcher) Start(server p2p.Server) error {
 		go f.handleRequest()
 	}
 
-	// subscribe chain head event
-	f.chainSub = f.blockchain.SubscribeChainEvent(f.chainCh)
+	if !f.config.NoDefaultStart {
+		// subscribe chain head event
+		f.chainSub = f.blockchain.SubscribeChainEvent(f.chainCh)
+		f.started = true
+	}
+
 	go f.reqLoop()
 	go f.resLoop()
 
 	return nil
 }
 
 func (f *ChainDataFetcher) Stop() error {
-	f.chainSub.Unsubscribe()
+	if f.chainSub != nil {
+		f.chainSub.Unsubscribe()
+	}
 	close(f.stopCh)
 
 	logger.Info("wait for all goroutines to be terminated...")
 	f.wg.Wait()
 	logger.Info("terminated all goroutines for chaindatafetcher")
+	f.started = false
+	return nil
+}
+
+func (f *ChainDataFetcher) start() error {
```

Comment:
`start` and `Start` methods confuse me. How about changing `start` to `startFetching`?
Also, I think `start` can be called inside of `Start`.

Reply:
How about start, stop -> startFetching, stopFetching / startRange, stopRange -> startRangeFetching, stopRangeFetching?

Reply:
Applied. df422607e7afac34e8b07c8daf66014632022461